src/JustEat.StatsD/Transport/Data.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+
+namespace JustEat.StatsD
+{
+    public ref struct Data
```

Comment:
@dv00d00 I'm confused, why is `Data` needed? It's a `ref struct` just like `ReadOnlySpan`, but it can take a `ReadOnlySpan` or a byte array, if you need to pass a byte array why not just wrap it in a `ReadOnlySpan`?

Reply:
Ah gotcha, I would have assumed that `ToArray` wouldn't copy in the case that the `Span` is backed by a heap array, but no, it copies as you say. :+1:

Reply:
Love this test that makes it very clear: https://github.com/dotnet/corefx/blob/master/src/System.Memory/tests/Span/ToArray.cs#L18